apollo-router-core/src/json_ext.rs

```diff
@@ -285,9 +285,7 @@ impl ValueExt for Value {
                 },
                 PathElement::Key(k) => match current_node {
                     Value::Object(o) => {
-                        current_node = o
-                            .get_mut(k.as_str())
-                            .expect("the value at that key was just inserted");
+                        current_node = o.entry(k.as_str()).or_insert(Value::default());
```

Comment:
That is not the same behaviour. If we ever get into a case where the value was not inserted we should get an error immediately, not silently update it, no?